src/transformers/generation/beam_search.py

```diff
@@ -156,6 +155,7 @@ def __init__(
         self,
         batch_size: int,
         num_beams: int,
+        max_length: int,
```

Comment:
BeamSearchScorer is a public class (even though, I don't think it's used that much on its own), do you think we could maybe do:

```suggestion
        max_length: Optional[int] = None,
```
and throw an error if `do_early_stopping` is set to `False` that says that one should `do_early_stopping=True` to not have to pass `max_length`.